packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -31,6 +35,105 @@ type BaseButtonWithIconProps = BaseButtonCommonProps & {
 
 export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
 
+const StyledBaseText = styled(BaseText)(
+  ({
+    hasIcon,
+    iconPosition,
+    iconSpacing,
+  }: BaseTextProps &
+    Pick<BaseButtonProps, 'iconPosition'> & { hasIcon: boolean; iconSpacing: string }) => `
+    padding-left: ${hasIcon && iconPosition === 'left' ? iconSpacing : 0};
+    padding-right: ${hasIcon && iconPosition === 'right' ? iconSpacing : 0};
+`,
+);
+
+type ConfigProps = {
+  iconSize: IconSize; // prop
+  fontSize: keyof Theme['typography']['fonts']['size']; // prop
+  lineHeight: keyof Theme['typography']['lineHeights']; // prop
+  buttonHeight: '48px' | '40px' | '32px' | '28px'; // raw value
+  iconSpacing: string; // raw value
+  iconColor: IconProps['color']; // prop
+  textColor: BaseTextProps['color']; // prop
+  buttonSpacing: string; // raw value
+  text?: string;
+};
+
+const getProps = ({
+  theme,
+  size,
+  children,
+}: {
+  theme: Theme;
+  size: BaseButtonCommonProps['size'];
+  children?: string;
+}): ConfigProps => {
+  const props: ConfigProps = {
+    iconSize: 'medium',
```

Comment:
We're following `small`, `medium`, etc for icon size & button size but `s`, `m`, etc. for `lineHeight`. Should We change one of them? Which one?

Reply:
I think we are also using `small`, `medium` for Typography so maybe better to change everything to `s`, `m`, `l` we can also change everything to `small`, `medium` but that might tricky when we reach `2xlarge`, `3xlarge` which looks weird in reading compared to `2xl`,`3xl`

Reply:
`sm`, `md`, `lg` looks weird. We'd then need `2xlg`, `3xlg`?. ` s`, `m`, `l` still gives you an idea of t-shirt sizing. My issue with all of these are they are not verbose enough. We don't have a character limit and `small`, `medium`, `large` looks pretty descriptive. Its like naming a variable `i` vs `index`. Closer it is to the English language, the more readable it is

Reply:
I think `xs, sm, md, lg, xl, 2xl` are standards today and widely acceptable because of [Bootstrap's grid system](https://getbootstrap.com/docs/4.0/layout/grid/) convention and [Tailwind](https://tailwindcss.com/docs/responsive-design). The t-shirt sizing logic makes sense, but yes they are not the most verbose nor readable. `small, medium & large` are readable but they are big prop names for what they do.

Reply:
+1 `xs, sm, md, lg, xl, 2lx` they are very standard names,  

Used by 
- [Chakra](https://chakra-ui.com/docs/components/form/button)
- [Carbon](https://carbondesignsystem.com/components/button/usage/)

The pattern which I'm seeing is component libraries which have only 3 variants for size mostly opt for `small`, `medium`, `large` (for eg: materialui, vercel) but if the variants are more than that xs, sm , md, lg.